core/src/Kokkos_View.hpp

```diff
@@ -19,13 +19,1008 @@
+  KOKKOS_FORCEINLINE_FUNCTION std::enable_if_t<
+      (Kokkos::Impl::always_true<I0, I1, I2, I3, Is...>::value && (4 == rank)),
+      reference_type>
+  access(I0 i0, I1 i1, I2 i2, I3 i3, Is... extra) const {
+    check_access_member_function_valid_args(extra...);
+    return base_t::operator()(i0, i1, i2, i3);
+  }
+
+  //------------------------------
+  // Rank 5
+
+  template <typename I0, typename I1, typename I2, typename I3, typename I4,
+            typename... Is>
+  KOKKOS_FORCEINLINE_FUNCTION std::enable_if_t<
+      (Kokkos::Impl::always_true<I0, I1, I2, I3, I4, Is...>::value &&
+       (5 == rank)),
+      reference_type>
+  access(I0 i0, I1 i1, I2 i2, I3 i3, I4 i4, Is... extra) const {
+    check_access_member_function_valid_args(extra...);
+    return base_t::operator()(i0, i1, i2, i3, i4);
+  }
+
+  //------------------------------
+  // Rank 6
+
+  template <typename I0, typename I1, typename I2, typename I3, typename I4,
+            typename I5, typename... Is>
+  KOKKOS_FORCEINLINE_FUNCTION std::enable_if_t<
+      (Kokkos::Impl::always_true<I0, I1, I2, I3, I4, I5, Is...>::value &&
+       (6 == rank)),
+      reference_type>
+  access(I0 i0, I1 i1, I2 i2, I3 i3, I4 i4, I5 i5, Is... extra) const {
+    check_access_member_function_valid_args(extra...);
+    return base_t::operator()(i0, i1, i2, i3, i4, i5);
+  }
+
+  //------------------------------
+  // Rank 7
+
+  template <typename I0, typename I1, typename I2, typename I3, typename I4,
+            typename I5, typename I6, typename... Is>
+  KOKKOS_FORCEINLINE_FUNCTION std::enable_if_t<
+      (Kokkos::Impl::always_true<I0, I1, I2, I3, I4, I5, I6, Is...>::value &&
+       (7 == rank)),
+      reference_type>
+  access(I0 i0, I1 i1, I2 i2, I3 i3, I4 i4, I5 i5, I6 i6, Is... extra) const {
+    check_access_member_function_valid_args(extra...);
+    return base_t::operator()(i0, i1, i2, i3, i4, i5, i6);
+  }
+
+  //------------------------------
+  // Rank 8
+
+  template <typename I0, typename I1, typename I2, typename I3, typename I4,
+            typename I5, typename I6, typename I7, typename... Is>
+  KOKKOS_FORCEINLINE_FUNCTION
+      std::enable_if_t<(Kokkos::Impl::always_true<I0, I1, I2, I3, I4, I5, I6,
+                                                  I7, Is...>::value &&
+                        (8 == rank)),
+                       reference_type>
+      access(I0 i0, I1 i1, I2 i2, I3 i3, I4 i4, I5 i5, I6 i6, I7 i7,
+             Is... extra) const {
+    check_access_member_function_valid_args(extra...);
+    return base_t::operator()(i0, i1, i2, i3, i4, i5, i6, i7);
+  }
+
+  //----------------------------------------
+  // Standard destructor, constructors, and assignment operators
+
+  KOKKOS_DEFAULTED_FUNCTION
+  ~View() = default;
+
+  KOKKOS_DEFAULTED_FUNCTION
+  View() = default;
+
+  KOKKOS_FUNCTION
+  View(const View& other) : base_t(other) {}
+
+  KOKKOS_FUNCTION
+  View(View&& other) : base_t(other) {}
```

Comment:
Don't we want to forward here?

Reply:
I think we should actually move here -- @crtrott any particular reason why it was written this way or okay to move?